tracer/src/Datadog.Trace/Tagging/TraceTagCollection.cs

```diff
@@ -144,26 +144,28 @@ public bool RemoveTag(string name)
                 ThrowHelper.ThrowArgumentNullException(nameof(name));
             }
 
-            var isPropagated = name.StartsWith(TagPropagation.PropagatedTagPrefix, StringComparison.OrdinalIgnoreCase);
+            var tags = _tags;
+            if (tags == null)
```

Comment:
Can we avoid taking the lock if `tags` is empty?
```suggestion
            if (tags == null || tags.Count == 0)
```

Reply:
done in [2262fee](https://github.com/DataDog/dd-trace-dotnet/pull/3907/commits/2262fee6703d64c3b08ae5087ccf8c97a23897dc)